binary/src/Cardano/Binary/FromCBOR.hs

```diff
@@ -184,7 +184,13 @@ instance FromCBOR Int64 where
   fromCBOR = D.decodeInt64
 
 instance (Integral a, FromCBOR a) => FromCBOR (Ratio a) where
-  fromCBOR = (%) <$ enforceSize "Ratio" 2 <*> fromCBOR <*> fromCBOR
+  fromCBOR = do
+    enforceSize "Ratio" 2
+    n <- fromCBOR
+    d <- fromCBOR
+    if d <= 0
```

Comment:
Are negative denominators also disallowed?

Reply:
I _think_ the canonical form for output has negatives in the numerator not the denominator.

Reply:
```
> 1 % (-1)
(-1) % 1
```